stdlib/test/builtin/test_list.mojo

```diff
@@ -38,7 +38,14 @@ fn test_repr_list() raises:
     assert_equal(__type_of(empty).__repr__(empty), "[]")
 
 
+def test_indexing():
+    var s = slice(1, 10)
+    assert_equal(s[True], 2)
+    assert_equal(s[UInt64(0)], 1)
```

Comment:
Here and in all the other tests: Could you please add one for just using an `IntLiteral`:
```mojo
    assert_equal(s[2], 3)
```